robot-server/robot_server/runs/run_store.py

```diff
@@ -296,11 +337,20 @@ def get_state_summary(self, run_id: str) -> Optional[StateSummary]:
             return (
                 json_to_pydantic(StateSummary, row.state_summary)
                 if row.state_summary is not None
-                else None
+                else BadStateSummary(
+                    dataError=InvalidStoredData(
+                        message="There was no engine state data for this run."
+                    )
+                )
             )
         except ValidationError as e:
             log.warning(f"Error retrieving state summary for {run_id}: {e}")
```

Comment:
Not related to your changes, but this improves the warning in case the stringification of `e` doesn't include enough information to debug. (It doesn't for `KeyError`s, for example.)

```suggestion
            log.warning(f"Error retrieving state summary for {run_id}", exc_info=True)
```